zokrates_core/src/semantics.rs

```diff
@@ -3755,10 +3772,9 @@ mod tests {
     fn function1() -> FunctionNode<'static> {
         let statements = vec![Statement::Return(None).mock()];
 
-        let arguments = vec![untyped::Parameter {
-            id: untyped::Variable::immutable("a", UnresolvedType::FieldElement.mock()).mock(),
-            is_private: true,
-        }
+        let arguments = vec![untyped::Parameter::public(
```

Comment:
Why is this change required?

Reply:
I found this inconsistent with other tests, we use the wrapper functions everywhere except here. The parameter must be public otherwise the test fails because of the check introduced by this PR.